src/SourceGenerators/Uno.UI.SourceGenerators.Internal/WeakEvent/WeakEventGenerator.cs

```diff
@@ -0,0 +1,154 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Uno.Extensions;
+using Uno.UI.SourceGenerators.Internal.Extensions;
+using Uno.UI.SourceGenerators.XamlGenerator;
+using Uno.UI.SourceGenerators.Helpers;
+using System.Linq;
+using System.Collections.Immutable;
+using System.Xml.Linq;
+
+namespace Uno.UI.SourceGenerators.Internal.WeakEvent;
+
+[Generator]
+public class WeakEventGenerator : IIncrementalGenerator
+{
+	private static SymbolDisplayFormat _fullyQualifiedWithoutGlobal = SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted);
+
+	private record GenerationCandidateData
+	{
+		public GenerationCandidateData(ISymbol symbol, AttributeData attribute)
+		{
+			var eventSymbol = symbol as IEventSymbol ?? throw new InvalidOperationException("Symbol must be an event");
+
+			var isDependencyObject = eventSymbol.ContainingType.AllInterfaces
+				.Any(t => t.ToDisplayString(_fullyQualifiedWithoutGlobal) == XamlConstants.Types.DependencyObject);
+
+			ContainingTypeIsCandidate = eventSymbol.ContainingType.TypeKind == TypeKind.Class && isDependencyObject;
```

Comment:
Any reason we limit generation to DependencyObject?

Reply:
At this time, there isn't any in code, but the assumption is that we may need to use loaded/unloaded specifics if performance becomes a concern with weak references. If we need to expand this for other reasons, we'll adjust.